clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -526,7 +544,7 @@ abstract class OpenstackClientProvider {
       client.useRegion(region).heat().stacks().getStackByName(stackName)
     }
     if (!stack) {
-      throw new OpenstackResourceNotFoundException("Unable to find stack $stackName in region $region")
+      throw new OpenstackProviderException("Unable to find stack $stackName in region $region")
```

Comment:
Do you explicitly cast `.toString()` in the constructor to whatever exception this subclasses? I just remembered that my custom Kubernetes exceptions originally failed because they expected a Java string, but got a Groovy string due to the `$templating` I use.